src/controllers/zendeskController.ts

```diff
@@ -63,28 +66,20 @@ export default class ZendeskController {
           r.kind === "IResponseSuccessJson" ? taskEither.of(r) : fromLeft(r)
         )
         .map(r => r.value)
-        .map(u => ({
-          fromProfileUser: u,
-          userEmail:
-            u.email && u.is_email_validated
-              ? u.email
-              : (u.spid_email as EmailString)
-        }))
-        .chain(({ fromProfileUser, userEmail }) =>
-          fromEither(
-            fromNullable(
-              ResponseErrorInternal("User does not have an email address")
-            )(userEmail)
-          ).map(email => ({ fromProfileUser, userEmail: email }))
+        .chain(fromProfileUser =>
+          fromPredicate(isProfileWithValidEmailAddress, _ =>
+            ResponseErrorInternal("User does not have an email address")
+          )(fromProfileUser)
         )
-        .chain(({ fromProfileUser, userEmail }) =>
+        .chain(profileWithValidEmailAddress =>
           this.tokenService
             .getJwtZendeskSupportToken(
               JWT_ZENDESK_SUPPORT_TOKEN_SECRET,
-              fromProfileUser.name as NonEmptyString,
-              fromProfileUser.family_name as NonEmptyString,
-              user.fiscal_code,
-              userEmail,
+              profileWithValidEmailAddress.name as NonEmptyString,
```

Comment:
What if `profileWithValidEmailAddress.name` is actually empty? I see two scenarios:
1. it doen't matter --> so we should relax the non-emptiness constrain on `getJwtZendeskSupportToken`
2. it's a problem --> a safer approach would be to validate values before the usage using a decode step, thus to handle error properly

Reply:
I'll go for 2. An empty name/surname would lead to a zendesk profile with empty name/surname.

Reply:
done in https://github.com/pagopa/io-backend/pull/847/commits/876ee601c6aee0f9492c0c94394997165d67af3b